bd_spot_wrapper/spot_wrapper/spot.py

```diff
@@ -863,6 +886,29 @@ def convert_transformation_from_BD_to_magnum(self, bd_transformation_dict: dict)
         mn_transformation = mn.Matrix4.from_(rotation_matrix, translation)
         return mn_transformation
 
+    def get_spot_a_T_b(self, a: str, b: str, tree=None) -> mn.Matrix4:
+        """
+        Gets transformation from 'a' frame to 'b' frame such that a_T_b
+        a & b takes string values of the name of the frames
+        tree is optional, its the kinematic transforms tree if none it will make new from robot's state
+        image sources also give us kinematic transforms trees that can be sent here
+        """
+        frame_tree_snapshot = (
+            self.get_robot_state().kinematic_state.transforms_snapshot
+            if tree is None
+            else tree
+        )
+        # BD api's function get_a_tform_b uses given transforms tree to make a_T_b
+        se3_pose = get_a_tform_b(frame_tree_snapshot, a, b)
+        # convert SE3Pose to magnum Matrix 4 transformation, seperate rotation & translation
+        pos = se3_pose.get_translation()
+        quat = se3_pose.rotation.normalize()
+        quat = quaternion.quaternion(quat.w, quat.x, quat.y, quat.z)
+        rotation_matrix = mn.Quaternion(quat.imag, quat.real).to_matrix()
+        translation = mn.Vector3(*pos)
+        mn_transformation = mn.Matrix4.from_(rotation_matrix, translation)
+        return mn_transformation
```

Comment:
We do not want to use multiple transformation libraries. sophuspy installation for our repo is easier than magnum.
We want to reduce usage on magnum.

Please replace this function with -
```
    def get_spot_a_T_b(self, frame_tree_snapshot: dict, a: str, b: str) -> sp.SE3:
        if frame_tree_snapshot is None:
            frame_tree_snapshot = (
                self.spot.get_robot_state().kinematic_state.transforms_snapshot
            )
        se3_pose = get_a_tform_b(frame_tree_snapshot, a, b)
        pos = se3_pose.get_translation()
        quat = se3_pose.rotation.normalize()
        return sp.SE3(quat.to_matrix(), pos)
```

Reply:
@KavitShah1998 Can you suggest alternative methods to vision_T_hand.transform_point(
        mn.Vector3(*point_in_local_3d)
    ) provided vision_T_hand is mn.Matrix4

Reply:
@jimmytyyang  It would be a significant change to port from magnum to sophuspy